artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -511,6 +512,10 @@ public QueueBinding updateQueue(SimpleString name,
             changed = true;
             queue.setExclusive(exclusive);
          }
+         if ((user != null && !user.equals(queue.getUser()) || (user == null && queue.getUser() != null))) {
+            changed = true;
+            queue.setUser(SimpleString.toSimpleString(user));
```

Comment:
Slight issue here in the logic that the older methods if invoked to update a queue will clear the user associated with the queue , as they will pass in null causing user to be set to null no matter.

Reply:
@jbertram ive sent a PR to address this https://github.com/apache/activemq-artemis/pull/2161